lib/AST/ASTImporter.cpp

```diff
@@ -6762,11 +6758,21 @@ Decl *ASTImporter::Import(Decl *FromD) {
 
   // Record the imported declaration.
   ImportedDecls[FromD] = ToD;
-  
+
   if (TagDecl *FromTag = dyn_cast<TagDecl>(FromD)) {
     // Keep track of anonymous tags that have an associated typedef.
     if (FromTag->getTypedefNameForAnonDecl())
       AnonTagsWithPendingTypedefs.push_back(FromTag);
+    else {
+
+      // Sanity check for structural equivalence of non-anonymous records.
+      // Currently, this will make the tests fail but can be very useful for
+      // debugging purposes.
+      //assert(!isa<RecordDecl>(FromD) ||
```

Comment:
Do we want to have this code here?

Reply:
Well, ok.
On a long term we want full structural equivalence between the original and the imported Decl. So perhaps the assertion would be elsewhere anyway, so we can get rid of this comment and the whole else block.